ArcGISRuntimeSDKQt_QMLSamples/Maps/ShowDeviceLocationUsingIndoorPositioning/ShowDeviceLocationUsingIndoorPositioning.qml

```diff
@@ -0,0 +1,198 @@
+    width: 800
+    height: 600
+
+    property FeatureTable positioningTable: null
+    property ArcGISFeatureTable pathwaysTable: null
+    property int currentFloor: -1
+    property var locationProperties: ({})
+
+    onPositioningTableChanged: setupIndoors()
+    onPathwaysTableChanged: setupIndoors()
+    onCurrentFloorChanged: changeFloorDisplay()
+
+    MapView {
+        id: mapView
+        anchors.fill: parent
+
+        Component.onCompleted: {
+            // Set and keep the focus on MapView to enable keyboard navigation
+            forceActiveFocus();
+        }
+
+        Map {
+            id: map
+            PortalItem {
+                itemId: "89f88764c29b48218366855d7717d266"
+                portal: Portal {
+                    url: "https://viennardc.maps.arcgis.com"
+                    credential: Credential {
+                        username: "tester_viennardc"
+                        password: "password.testing12345"
+                    }
+                }
+            }
+
+            onLoadStatusChanged: {
+                if (loadStatus !== Enums.LoadStatusLoaded)
+                    return;
+
+                findPositioningTable("ips_positioning")
+                findPathwaysTable("Pathways");
+            }
+        }
+    }
+
+    Rectangle {
+        id: debugInfoRectangle
+        width: parent.width
+        height: textColumn.height + 20
+        color: "lightgray"
+
+        visible: false
+
+        ColumnLayout {
+            id: textColumn
+            anchors {
+                top: parent.top
+                left: parent.left
+                margins: 10
+            }
+
+            Text {
+                id: floorText
+                visible: false
+            }
+            Text {
+                id: positionSourceText
+                visible: false
+            }
+            Text {
+                id: transmitterCountText
+                visible: false
+            }
+            Text {
+                id: horizontalAccuracyText
+                visible: false
+            }
+        }
+    }
+
+    // An IPS positioning feature table is stored with an IPS-enabled map. Each row in the table contains an indoor positioning file.
+    // The IndoorsLocationDataSource will use the most recently created row unless given an alternative GlobalId in the constructor.
+    function findPositioningTable(tableName) {
+        map.tables.forEach((table) => {
+                               table.loadStatusChanged.connect(() => {
+                                                                   if (table.loadStatus === Enums.LoadStatusLoaded && table.tableName === tableName) {
+                                                                       positioningTable = table;
+                                                                   }
+                                                               });
+                               table.load();
+                           });
```

Comment:
This looks _correct_, but I'd recommend easing off on the indenting. This is always tough to nail down how much and what is correct.